app/src/main/kotlin/com/wire/android/ui/home/HomeScreen.kt

```diff
@@ -45,185 +44,211 @@ import com.wire.android.ui.common.dimensions
+                            // we want to render "nothing" instead of doing a if/else check
+                            // on homeBottomSheetContent and wrap homeContent() into WireModalSheetLayout
+                            // or render it without WireModalSheetLayout to avoid
+                            // recomposing the homeContent() when homeBottomSheetContent
+                            // changes from null to "something"
+                            sheetContent = homeBottomSheetContent ?: { }
+                        ) {
+                            CollapsingTopBarScaffold(
+                                snapOnFling = false,
+                                keepElevationWhenCollapsed = true,
+                                topBarHeader = { elevation ->
+                                    Column(modifier = Modifier.animateContentSize()) {
+                                        CommonTopAppBar(
+                                            connectivityUIState = connectivityState,
+                                            onReturnToCallClick = onReturnToCallClick
+                                        )
+                                        AnimatedVisibility(visible = !searchBarState.isSearchActive) {
+                                            HomeTopBar(
+                                                avatarAsset = avatarAsset,
+                                                status = status,
+                                                title = stringResource(currentNavigationItem.title),
+                                                elevation = elevation,
+                                                onHamburgerMenuClick = ::openDrawer,
+                                                onNavigateToSelfUserProfile = onSelfUserClick
+                                            )
+                                        }
+                                    }
+                                },
+                                snackbarHost = {
+                                    SwipeDismissSnackbarHost(
+                                        hostState = snackBarHostState,
+                                        modifier = Modifier.fillMaxWidth()
+                                    )
+                                },
+                                topBarCollapsing = {
+                                    if (currentNavigationItem.isSearchable)
+                                        SearchTopBar(
+                                            isSearchActive = searchBarState.isSearchActive,
+                                            searchBarHint = stringResource(
+                                                R.string.search_bar_hint,
+                                                stringResource(id = title).lowercase()
+                                            ),
+                                            searchQuery = searchBarState.searchQuery,
```

Comment:
There is another screen being rendered if you mean NavigationItem i do not think it makes sense in this case because we would need to fetch all the conversation inside the viewmodel for that navigationitem here we can reuse what we already have, however on next iteration we need to perform sql query and it will make sense to use a navigationitem then I think

Reply:
We can create a cache to reuse the data on both screens, but ok, it's fine now.